scheduler/common/cluster-client/src/main/kotlin/ac/uk/ebi/cluster/client/lsf/ClusterOperations.kt

```diff
@@ -36,18 +35,12 @@ class ClusterOperations(
 
     companion object {
 
-        private val sshClient = JSch()
         private val responseParser = JobResponseParser()
 
-        fun create(user: String, password: String?, sshMachine: String): ClusterOperations {
-            return ClusterOperations(responseParser) { createSession(user, password, sshMachine) }
-        }
-
-        private fun createSession(user: String, password: String?, sshMachine: String): Session {
-            val session = sshClient.getSession(user, sshMachine)
-            session.setConfig("StrictHostKeyChecking", "no")
-            password?.let { session.setPassword(it) }
-            return session
+        fun create(sshKey: String, sshMachine: String): ClusterOperations {
+            val sshClient = JSch()
+            sshClient.addIdentity(sshKey)
```

Comment:
I noticed that, at some point, you stopped using `apply`. Is there any reason for that?

Reply:
> I noticed that, at some point, you stopped using `apply`. Is there any reason for that?

I think is usefull but we need to also consider that someone need to remember what the apply function does to understand code so I am using more carefully, when really need it.